src/Contrib/Otlp/MetricExporterFactory.php

```diff
@@ -67,6 +68,21 @@ private function buildTransport(string $protocol): TransportInterface
         );
     }
 
+    private function getTemporality(): ?string
+    {
+        $value = Configuration::getEnum(Variables::OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE);
+        switch (strtolower($value)) {
+            case 'cumulative':
+                return 'cumulative';
+            case 'delta':
+                return 'delta';
+            case 'lowmemory':
+                return null;
+            default:
+                throw new \UnexpectedValueException('Unknown temporality: ' . $value);
+        }
+    }
```

Comment:
```suggestion
    private function getTemporality()
    {
        $value = Configuration::getEnum(Variables::OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE);
        switch (strtolower($value)) {
            case 'cumulative':
                return Temporality::CUMULATIVE;
            case 'delta':
                return Temporality::DELTA;
            case 'lowmemory':
                return null;
            default:
                throw new \UnexpectedValueException('Unknown temporality: ' . $value);
        }
    }
```

Reply:
The correct typehint would be `string|Temporality|null`, union types are >= 8.0.